src/main/java/com/joy/joyfillandroid/Tablefield/MobileSDKField.kt

```diff
@@ -0,0 +1,401 @@
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.IOException
+
+@RequiresApi(Build.VERSION_CODES.P)
+class MobileSDKField(context: Context) : LinearLayout(context){
+    val componentLayout = LinearLayout(context)
+    var yValues_mobile = ArrayList<Int>()
+    var component_type = ArrayList<String>()
+    var Sequence_With_Y_Position: List<String> = emptyList()
+    var dataValues = ArrayList<String>()
+    var tableColumnOrderArrayID = ArrayList<String>()
+    val tableColumnType = mutableListOf<String>()
+    val tableColumnTitle = mutableListOf<String>()
+    val tableColumnValue = mutableListOf<String>()
+    var type: String? = null
+    var list= ArrayList<componentModel>()
+    val mainHandler = Handler(context.mainLooper)
+    var options: JSONArray? = null
+    private val client = OkHttpClient()
+    object SharedData {
+        var joyfillDocID: String = ""
+        var authorizedToken: String = ""
+    }
+    val textField = TextField(context)
+    val dropDown = DropDown(context)
+    val multipleChoice = MultipleChoice(context)
+
+
+    val typePositionsMap = mutableMapOf<String, Int>()
+    val idPositionsMap = mutableMapOf<String, Int>()
+
+    init {
+        orientation = VERTICAL
+        val layoutParams = LayoutParams(
+            LayoutParams.MATCH_PARENT,
+            LayoutParams.MATCH_PARENT
+        )
+        setLayoutParams(layoutParams)
+
+        layoutParams.setMargins(dpToPx(context, 8), dpToPx(context, 8), dpToPx(context, 8), dpToPx(context, 8))
+        val componentLayoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        componentLayout.orientation = LinearLayout.VERTICAL
+        componentLayout.layoutParams = componentLayoutParams
+
+        val view = LinearLayout(context)
+        val viewLayoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT,
+        )
+
+        view.layoutParams = viewLayoutParams
+        view.orientation = VERTICAL
+        view.background = layoutBackground()
+
+        val scrollView = NestedScrollView(context)
+        val scrollViewParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+        scrollViewParams.setMargins(0, dpToPx(context, 10), 0, dpToPx(context, 10))
+        scrollView.layoutParams= scrollViewParams
+        scrollView.setVerticalScrollBarEnabled(false)
+        scrollView.overScrollMode= View.OVER_SCROLL_NEVER
+        scrollView.fullScroll(ScrollView.FOCUS_DOWN)
+
+        scrollView.addView(componentLayout)
+        view.addView(scrollView)
+        addView(view)
+        makeApiCall()
+    }
+
+    //Mark: Function for API Call
+    fun makeApiCall() {
```

Comment:
I would name this something more like `retrieveJoyfillData` or something with comments throughout explaining how it all works

Reply:
Done